src/dune_pkg/solver_env.ml

```diff
@@ -158,35 +209,41 @@ end
 
 type t =
   { sys : Variable.Sys.Bindings.t
+  ; user : Variable.User.Bindings.t
   ; const : Variable.Const.Bindings.t
   }
 
 module Fields = struct
   let sys = "sys"
+  let user = "user"
   let const = "const"
 end
 
-let create ~sys = { sys; const = Variable.Const.bindings }
-let default = create ~sys:Variable.Sys.Bindings.empty
+let create ~sys ~user = { sys; user; const = Variable.Const.bindings }
+let default = create ~sys:Variable.Sys.Bindings.empty ~user:Variable.User.Map.empty
 
 let decode =
   let open Decoder in
   fields
   @@
-  let+ sys = field Fields.sys ~default:default.sys Variable.Sys.Bindings.decode in
+  let+ sys = field Fields.sys ~default:default.sys Variable.Sys.Bindings.decode
```

Comment:
No need to keep this field as separate. Users don't need to care about this distinction. They can set/override all variables.